src/createSchemaForm.tsx

```diff
@@ -392,58 +397,64 @@ export function createTsForm<
       });
     }
     const submitFn = handleSubmit(_submit);
-    return (
-      <FormProvider {..._form}>
-        <ActualFormComponent {...formProps} onSubmit={submitFn}>
-          {renderBefore && renderBefore({ submit: submitFn })}
-          {Object.keys(shape).map((key) => {
-            const type = shape[key] as RTFSupportedZodTypes;
-            const Component = getComponentForZodType(type, componentMap);
-            if (!Component) {
-              throw new Error(
-                noMatchingSchemaErrorMessage(key, type._def.typeName)
-              );
-            }
-            const meta = getMetaInformationForZodType(type);
+    type SchemaKey = keyof z.infer<UnwrapEffects<SchemaType>>;
+    const renderedFields = Object.keys(shape).reduce((accum, key : SchemaKey) => {
```

Comment:
sorry this diff is so gross cause everything got un-indented! basically I just pulled out mapped fields here into an object mapped by schema key.

Reply:
hide whitespace makes this much more readable fyi